polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala

```diff
@@ -324,10 +327,45 @@ object SocketTransport {
       }
     }
 
+    private def findScalaVersion: URIO[BaseEnv with GlobalEnv with CurrentNotebook, String] = for {
+      serverConfig   <- Config.access
+      notebookConfig <- CurrentNotebook.config
+      scalaVersion   <- ZIO.succeed(notebookConfig.scalaVersion).some
+        .orElse(ZIO.succeed(serverConfig.kernel.scalaVersion).some)
+        .orElse(deployCommand.detectScalaVersion.some)
```

Comment:
What do you think about potentially adding the Scala version to `SparkPropertySet`? If Scala versions vary by property that seems like a logical place to specify that (and could solve possible issues if `spark-submit` version detection doesn't work for some reason.

Reply:
That's a good point! I agree it's a natural place to configure the Scala version.